std/typecons.d

```diff
@@ -422,19 +422,31 @@ template Tuple(Specs...)
         U u = U.init;
         T t = u;
     }));
+
     /+ Helper for partial instanciation +/
     template isBuildableFrom(U)
     {
         enum isBuildableFrom(T) = isBuildable!(T, U);
     }
 
+    /+ Determine whether or not all given types in T are the same type +/
+    template areAllSame(T...)
+    {
+        static if (T.length == 0 || T.length == 1)
```

Comment:
That's more likely to raise a brow with most maintainers than the simpler condition, so please change and let's merge this. I approve the PR and will archive this.